mesheryctl/internal/cli/root/root.go

```diff
@@ -58,50 +60,68 @@ func (f *TerminalFormatter) Format(entry *log.Entry) ([]byte, error) {
 	return append([]byte(entry.Message), '\n'), nil
 }
 
-// rootCmd represents the base command when called without any subcommands
-var rootCmd = &cobra.Command{
+var (
+	availableSubcommands = []*cobra.Command{}
+)
+
+// RootCmd represents the base command when called without any subcommands
+var RootCmd = &cobra.Command{
 	Use:   "mesheryctl",
 	Short: "Meshery Command Line tool",
 	Long:  `Meshery is the service mesh management plane, providing lifecycle, performance, and configuration management of service meshes and their workloads.`,
+	Args:  cobra.MinimumNArgs(1),
 	// Uncomment the following line if your bare application
 	// has an action associated with it:
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		b, _ := cmd.Flags().GetBool("version")
 		if b {
 			versionCmd.Run(nil, nil)
-			return
+			return nil
 		}
-		if len(args) == 0 {
-			log.Print(cmdDetails)
+
+		for _, subcommand := range availableSubcommands {
+			if args[0] == subcommand.Name() {
+				return nil
+			}
 		}
+
+		return errors.New("sub-command not found : " + "\"" + args[0] + "\"")
 	},
 }
 
 // Execute adds all child commands to the root command and sets flags appropriately.
-// This is called by main.main(). It only needs to happen once to the rootCmd.
+// This is called by main.main(). It only needs to happen once to the RootCmd.
 func Execute() {
 	//log formatter for improved UX
 	log.SetFormatter(new(TerminalFormatter))
-	if err := rootCmd.Execute(); err != nil {
+	if err := RootCmd.Execute(); err != nil {
```

Comment:
When a user doesn't include an argument, but only executes `mesheryctl`, the help text is show, plus this error message: `fatal err requires at least 1 arg(s), only received 0`. It is shown twice - once at the beginning and once at the end of the output returned. 

```
Error: requires at least 1 arg(s), only received 0
Usage:
  mesheryctl [flags]
  mesheryctl [command]

Available Commands:
  help        Help about any command
  perf        Performance Testing
  system      Meshery Lifecycle Management
  version     Version of mesheryctl

Flags:
      --config string   config file (default location is: /Users/lee/.meshery/meshery.yaml)
  -h, --help            help for mesheryctl
  -v, --version         Version flag

Use "mesheryctl [command] --help" for more information about a command.

fatal err requires at least 1 arg(s), only received 0
```

Prior behavior only showed the user a copy of the help text. Could we revert to this behavior?

Reply:
I can do that. The reason shows the error now is because I enabled the MinArguments flag in each of the commands. Will revert and push soon

Reply:
@leecalcote Removed the redundant error message that was being returned because of handling errors in `cmd.Execute`.

Reply:
@nitishm thank you. The duplicative error message is now gone. Are we able to remove the other error message as well (fully revert) and only show the user the help / usage text instead? Other related CLIs like `docker`, `kubectl`, and `istioctl` also skip showing an error message, but show the command-specific help text instead.

Reply:
Let me see what I can do. Might need to remove the num args enforcement I think